test/test-tab.js

```diff
@@ -127,12 +127,6 @@ exports["test behavior on close"] = function(assert, done) {
                      "After being closed, tab attributes are undefined (url)");
         assert.equal(tab.index, undefined,
                      "After being closed, tab attributes are undefined (index)");
-        if (app.is("Firefox")) {
-          // Ensure that we can call destroy multiple times without throwing;
-          // Fennec doesn't use this internal utility
-          tab.destroy();
-          tab.destroy();
-        }
```

Comment:
As far as I can see from the existing implementation (there are no docs), calling destroy on a tab effectively hides it from the tabs APIs. It will no longer be visible in the tabs list nor will any events be fired for it. I couldn't think of a single reason for doing that and it seems like a footgun to me. We also have no testing of it other than this test that we can call destroy. I could try to reimplement the behaviour but since the Fennec API doesn't implement this either I think it makes more sense to drop it. I could add a no-op method too I guess.